dashboard/src/actions/authActions.js

```diff
@@ -7,6 +7,161 @@ import Cookies from "js-cookie";
 import { SUCCESS } from "assets/constants/overviewConstants";
 import { showToast } from "actions/toastActions";
 import { uid } from "../utils/helper";
+import { fetchEndpoints } from "actions/endpointAction";
+
+
+/**
+ * Get the key value.
+ * @param {key} key can be a string.
+ * @return {value} a string value of a key
+ */
+export function getQueryVariable (key) {
+    const query = window.location.search.substring(1);
+    const vars = query.split('&');
+    let code = "";
+    for (let i = 0; i < vars.length; i++) {
+      const pair = vars[i].split('=');
+      if (pair[0] === key) {
+        code = pair[1];
+      }
+      // else return "no-idea";
+    }
+    return code;
+
+}
+
+/**
+ * Get user details from the OIDC server.
+ * @param {endpoints} endpoints from the getState.
+ * @return {user} a dict value of a user details obtained from the token
+ */
+export function getOIDCUserDetails(endpoints) {
+    const reqUser = new XMLHttpRequest();
+    const OIDCToken = Cookies.get("token");
+    const oidcServer = endpoints.openid.server;
+    const oidcRealm = endpoints.openid.realm;
+    const uri = `${oidcServer}/realms/${oidcRealm}/protocol/openid-connect/userinfo`;
```

Comment:
This sequence occurs often enough that we might want to consider encapsulating and even caching it somehow.

E.g., instead of fetching `endpoints.openid.server` and `endpoints.openid.realm`, suppose we fetched `endpoints.uri_base` which returned `${oidcServer}/realms/${oidcRealm}/protocol/openid-connect` already assembled.  (We might not want to store it under `endpoints`, but that might be an OK place -- that is, I'm not saying that the Pbench Server should send the constructed URI, although that might be worth considering, as well.)